napi-inl.h

```diff
@@ -42,6 +41,9 @@ namespace details {
 
 #endif // NAPI_CPP_EXCEPTIONS
 
+#define NAPI_FATAL_IF_FAILED(status, location, message) \
+  if ((status) != napi_ok) Error::Fatal((location), (message));
```

Comment:
Suggestion: wrap this in a `do { ... } while` so that `if (...) NAPI_FATAL_IF_FAILED(...); else ...` does the right thing.